src/unity/lib/unity_sframe.cpp

```diff
@@ -212,6 +212,8 @@ std::map<std::string, std::shared_ptr<unity_sarray_base>> unity_sframe::construc
   if (csv_parsing_config["quote_char"].get_type() == flex_type_enum::STRING) {
     std::string tmp = (flex_string)csv_parsing_config["quote_char"];
     if (tmp.length() > 0) tokenizer.quote_char = tmp[0];
+  } else if (csv_parsing_config["quote_char"].get_type() == flex_type_enum::UNDEFINED) {
```

Comment:
hmm. This is not the ideal fix. tokenizer.quote_char is a char, so this really has the effect is setting the quote char to '\0'. 

Technically the current csv parser implementation does not support the disabling of quote_char. In a way, setting to '\0' is kind of a workaround.

Reply:
I will take a look at supporting the disabling of quote_char as an parsing option after I get #1737 in since that set of changes will most certainly conflict.

Reply:
Ok, we can wait till after #1737 is merged. Setting `quote_char='\0'` in Python does solve the problem. However `quote_char=None` just makes a lot more sense in Python; that's how I tried to solve the problem and was surprised when it didn't work.

Reply:
#1734 has been merged. I've rebased this PR on top of master. Everything passes.